src/components/utils/src/threads/thread_posix.cc

```diff
@@ -86,12 +86,16 @@ void* Thread::threadFunc(void* arg) {
 
   pthread_cleanup_push(&cleanup, thread);
 
-  thread->state_lock_.Acquire();
+  {
+    sync_primitives::AutoLock auto_lock(thread->run_lock_);
```

Comment:
@SNiukalov Lock for acquiring another lock?

Reply:
This is not a lock for acquiring another lock. This code part ensures correct notification sequence between Thread::start and threadFunc. In Thread::start function after thread is created, we call Wait(), but we cannot guarantee order of precedence of the following calls : "Wait() in Thread::start" or "Broadcast in threadFunc". sync_primitives::AutoLock auto_lock(thread->run_lock_); - is used to guarantee call to "Wait() in Thread::start" before Broadcast.